### PR TITLE
Misc `ReferenceDocs` improvements

### PR DESCRIPTION
**Pull request by DaveCTurner**
- Forbid ephemeral `_auto_gen.html` and `page.html#_auto_gen` links.
- Remove dangling/unused `BOOTSTRAP_CHECK_G1GC` link.
- Separate test suite into individual tests.

### CODE REVIEW

File: docs/reference/setup/bootstrap-checks-xes.asciidoc [hunk 1]
```diff
@@ -6,6 +6,7 @@ In addition to the <<bootstrap-checks,{es} bootstrap checks>>, there are
 checks that are specific to {xpack} features.
 
 [discrete]
+[[bootstrap-checks-xpack-encrypt-sensitive-data]]
```
Comment (PhilYue) [inline docs/reference/setup/bootstrap-checks-xes.asciidoc:9]: test

**Comment by DaveCTurner** [comment]
Note that we're not currently validating these links anyway, see https://github.com/elastic/docs/pull/2656

**Comment by rjernst** [comment]
Why yaml? I had actually made a change to the docs build to support tab delimited txt, but lost track of the corresponding change in ES. My intention there was to be able to move reference docs into the core/lib, so it could be used in other libs. As it is now, it can't because it requires xcontent.

**Comment by DaveCTurner** [comment]
> Why yaml?

Only because it is easy to parse correctly and has no interspersed separators (i.e. no trailing commas). But I can move this to an XContent-free format too if you'd prefer. I'll remove the format change from this PR and open a follow-up.

**Review by rjernst — APPROVED** [review]
LGTM

**Pull request merged**